src/simulation/LoadGenerator.cpp

```diff
@@ -843,265 +619,68 @@ LoadGenerator::TxMetrics::report()
-
-    case TxInfo::TX_TRANSFER_NATIVE:
         txm.mPayment.Mark();
         txm.mNativePayment.Mark();
-        txs.push_back(txtest::createPaymentTx(app, mFrom->mKey,
-                                              mTo->mKey.getPublicKey(),
-                                              mFrom->mSeq + 1, mAmount));
-        break;
-
-    case TxInfo::TX_TRANSFER_CREDIT:
-    {
-        txm.mPayment.Mark();
-        std::vector<Asset> assetPath;
-        for (auto acc : mPath)
-        {
-            assert(!acc->mIssuedAsset.empty());
-            assetPath.emplace_back(
-                txtest::makeAsset(acc->mKey, acc->mIssuedAsset));
-        }
-        assert(!assetPath.empty());
-        if (assetPath.size() == 1)
-        {
-            txm.mCreditPayment.Mark();
-            txs.emplace_back(txtest::createCreditPaymentTx(
-                app, mFrom->mKey, mTo->mKey.getPublicKey(), assetPath.front(),
-                mFrom->mSeq + 1, mAmount));
-        }
-        else
-        {
-            switch (assetPath.size())
-            {
-            case 2:
-                txm.mOneOfferPathPayment.Mark();
-                break;
-            case 3:
-                txm.mTwoOfferPathPayment.Mark();
-                break;
-            default:
-                txm.mManyOfferPathPayment.Mark();
-                break;
-            }
-
-            auto sendAsset = assetPath.front();
-            auto recvAsset = assetPath.back();
-            assetPath.erase(assetPath.begin());
-            assetPath.pop_back();
-            auto sendMax = mAmount * 10;
-            auto fromAccount = TestAccount{app, mFrom->mKey};
-            auto op =
-                txtest::pathPayment(mTo->mKey.getPublicKey(), sendAsset,
-                                    sendMax, recvAsset, mAmount, assetPath);
-            txs.emplace_back(fromAccount.tx({op}, mFrom->mSeq + 1));
-        }
     }
-    break;
+    txm.mTxnAttempted.Mark();
 
-    default:
-        assert(false);
+    {
+        StellarMessage msg;
+        msg.type(TRANSACTION);
+        msg.transaction() = txf->getEnvelope();
+        txm.mTxnBytes.Mark(xdr::xdr_argpack_size(msg));
     }
-}
 
-void
-LoadGenerator::TxInfo::recordExecution(int64_t baseFee)
-{
-    mFrom->mSeq++;
-    mFrom->mBalance -= baseFee;
-    if (mFrom && mTo)
+    auto status = app.getHerder().recvTransaction(txf);
+    if (status != Herder::TX_STATUS_PENDING)
     {
-        if (!mPath.empty())
-        {
-            for (auto& tl : mFrom->mTrustLines)
-            {
-                if (tl.mIssuer == mPath.front())
-                {
-                    tl.mBalance -= mAmount;
-                }
-            }
-            for (auto& tl : mTo->mTrustLines)
-            {
-                if (tl.mIssuer == mPath.back())
-                {
-                    tl.mBalance += mAmount;
-                }
-            }
-        }
-        else
+        static const char* TX_STATUS_STRING[Herder::TX_STATUS_COUNT] = {
```

Comment:
it is duplicated in CommandHandler, I think we need to extract that to somewhere